packages/expo-notifications/src/getDevicePushTokenAsync.web.ts

```diff
@@ -0,0 +1,111 @@
+import { CodedError, Platform, SyntheticPlatformEmitter } from '@unimodules/core';
+import Constants from 'expo-constants';
+
+import { DevicePushToken } from './getDevicePushTokenAsync';
+
+export default async function getDevicePushTokenAsync(): Promise<DevicePushToken> {
+  const data = await _subscribeUserToPushAsync();
+  SyntheticPlatformEmitter.emit('onDevicePushToken', { devicePushToken: data });
+  return { type: Platform.OS, data };
+}
+
+function guardPermission() {
+  if (!('Notification' in window)) {
+    throw new Error('The Notification API is not available on this device.');
```

Comment:
This should be a coded error so developers can handle it and know there isn't an issue with their own code:
```
throw new CodedError('ERR_UNAVAILABLE', `The Web Notifications API is not available on this device`);
```